File: Runners/JavaRunner/java-runner/src/main/java/com/mems/manifest/SampleTest.java
// SampleTest.java
package com.mems.manifest;

import com.fasterxml.jackson.databind.JsonNode;

public class SampleTest {
    public String name;
    public JsonNode inputs;    // arbitrary JSON
    public JsonNode expected;  // arbitrary JSON
    public String comparator;  // eq, neq, seq_eq, seq_eq_sorted, contains, lt, gt, le, ge
    public long timeoutMs;
}

File: Runners/KotlinRunner/src/main/kotlin/manifest/SampleTest.kt
// SampleTest.kt
package manifest

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.JsonNode

data class SampleTest(
    @field:JsonProperty("Name")
    var name: String? = null,
    @field:JsonProperty("Inputs")
    var inputs: JsonNode? = null,    // arbitrary JSON
    @field:JsonProperty("Expected")
    var expected: JsonNode? = null,  // arbitrary JSON
    @field:JsonProperty("Comparator")
    var comparator: String? = null,  // eq, neq, seq_eq, seq_eq_sorted, contains, lt, gt, le, ge
    @field:JsonProperty("TimeoutMs")
    var timeoutMs: Long = 0L
)
